sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +135,277 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
+
+      // If not unsuccessful weeks, then clone new offering and create re-assessment.
+      const notUnsuccessfulWeeksScholasticStanding =
+        scholasticStandingData.scholasticStanding !==
+          SCHOLASTIC_STANDING_STUDENT_DID_NOT_COMPLETE_PROGRAM &&
+        !scholasticStandingData.numberOfUnsuccessfulWeeks;
+
+      if (notUnsuccessfulWeeksScholasticStanding) {
+        // Cloning existing offering.
+        const offering: EducationProgramOffering = { ...existingOffering };
+
+        // Assigning id and audit fields as undefined, so that when its saved its considered as a new EducationProgramOffering object.
+        offering.id = undefined;
+        offering.createdAt = undefined;
+        offering.updatedAt = undefined;
+        offering.creator = auditUser;
+        offering.modifier = auditUser;
+
+        const newStudyEndDate =
+          scholasticStandingData.dateOfChange ??
+          scholasticStandingData.dateOfCompletion ??
+          scholasticStandingData.dateOfWithdrawal;
+
+        offering.studyEndDate = new Date(newStudyEndDate);
+
+        offering.actualTuitionCosts =
+          scholasticStandingData.tuition ?? existingOffering.actualTuitionCosts;
+        offering.programRelatedCosts =
+          scholasticStandingData.booksAndSupplies ??
+          existingOffering.programRelatedCosts;
+        offering.mandatoryFees =
+          scholasticStandingData.mandatoryFees ??
+          existingOffering.mandatoryFees;
+        offering.exceptionalExpenses =
+          scholasticStandingData.exceptionalCosts ??
+          existingOffering.exceptionalExpenses;
+
+        offering.offeringType = OfferingTypes.ScholasticStanding;
+
+        // Save new offering.
+        const savedOffering = await transactionalEntityManager
+          .getRepository(EducationProgramOffering)
+          .save(offering);
+
+        // Create the new assessment to be processed.
+        scholasticStanding.studentAssessment = {
+          application: { id: application.id } as Application,
+          triggerType: AssessmentTriggerType.ScholasticStandingChange,
+          creator: auditUser,
+          submittedBy: auditUser,
+          submittedDate: now,
+          offering: { id: savedOffering.id } as EducationProgramOffering,
+        } as StudentAssessment;
+      } else {
+        // If unsuccessful weeks, then add to the column.
+        // * No cloning of offering and re-assessment is required in this scenario.
+        scholasticStanding.unsuccessfulWeeks =
+          scholasticStandingData.numberOfUnsuccessfulWeeks;
+      }
 
       const studentScholasticStanding = await transactionalEntityManager
         .getRepository(StudentScholasticStanding)
         .save(scholasticStanding);
 
-      // Save current application.
-      application.currentAssessment = {
-        id: scholasticStanding.studentAssessment.id,
-      } as StudentAssessment;
+      if (scholasticStanding.studentAssessment) {
```

Comment:
Is there anything that prevents us from moving this code right after line 214 inside the existing `if`?

Reply:
yes,  in order to get the `scholasticStanding.studentAssessment.id` scholastic standing must be saved the saving is happening just above `222`,

Reply:
I git it now, thanks!